src/Symfony/Bridge/Twig/Tests/Extension/HttpFoundationExtensionTest.php

```diff
@@ -41,6 +41,14 @@ public function getGenerateAbsoluteUrlData()
             array('http://example.com/baz', 'http://example.com/baz', '/'),
             array('https://example.com/baz', 'https://example.com/baz', '/'),
             array('//example.com/baz', '//example.com/baz', '/'),
+
+            array('http://localhost/foo/bar?baz', '?baz', '/foo/bar'),
+            array('http://localhost/foo/bar?baz=1', '?baz=1', '/foo/bar?foo=1'),
+            array('http://localhost/foo/baz?baz=1', 'baz?baz=1', '/foo/bar?foo=1'),
+
+            array('http://localhost/foo/bar#baz', '#baz', '/foo/bar'),
+            array('http://localhost/foo/bar?baz=1#baz', '?baz=1#baz', '/foo/bar?foo=1'),
+            array('http://localhost/foo/baz?baz=1#baz', 'baz?baz=1#baz', '/foo/bar?foo=1'),
         );
```

Comment:
Ah this is only executed when no current request is given (CLI) and the request context is used instead :wink: 

So your changes from L 73-79 are currently not covered by tests. You need to add some more test cases into the `getGenerateAbsoluteUrlRequestContextData` provider :wink: